src/net_builtin.c

```diff
@@ -552,17 +552,19 @@ long mg_io_send(struct mg_connection *c, const void *buf, size_t len) {
   struct mg_tcpip_if *ifp = (struct mg_tcpip_if *) c->mgr->priv;
   struct connstate *s = (struct connstate *) (c + 1);
   uint32_t rem_ip;
+  size_t eth_h_len = 14, ip_max_h_len = 24, tcp_max_h_len = 60, udp_h_len = 8;
+  size_t max_headers_len = eth_h_len + ip_max_h_len +
+                          (c->is_udp ? udp_h_len : tcp_max_h_len);
   memcpy(&rem_ip, c->rem.ip, sizeof(uint32_t));
+  if (len + max_headers_len > ifp->tx.len) {
+    len = ifp->tx.len - max_headers_len;
+  }
+  if (len + max_headers_len - eth_h_len > c->mtu) {
```

Comment:
Also, `mtu` must not be too small - it should be bigger that the size of headers. Shall we check that `mtu` is not less than a minimum possible here, or somewhere else?